### DISCUSSION TITLE
중간 배포 오류 일지

### DISCUSSION THREAD

**Discussion by punchdrunkard** [prgrms-web-devcourse/Team-BackFro-ArtZip-FE#105] [00.TroubleShooting]
- 관련 이슈 번호 : #79 

# 공통적인 에러

- `key` props 정의가 안되어있을 때
- 타입 정의가 잘못 되었을 때 (**타입 스크립트 관련 에러가 정말 많이 남!!**)

우선은 주석 처리와 같은 방법으로 막아두었지만,
앞으로 배포를 위하여 husky를 더 잘 이용할 필요가 있어보입니다ㅠㅠ

![image](https://user-images.githubusercontent.com/74234333/182877261-9d6a9658-def7-4882-ae8d-e688b182a721.png)

# style 이 `page` 안에 있는 폴더에 함께 있는 경우

![image](https://user-images.githubusercontent.com/74234333/182881303-aea351dc-3af1-4a7c-bffc-8eb0267a5695.png)
위 사진처럼 `pages` 아래에 폴더가 선언되어있을 경우 나타나는 에러입니다.

![image](https://user-images.githubusercontent.com/74234333/182880573-4daa2d51-61ad-40d6-9275-7d2ac2fb72f1.png)

- 에러 메시지 : `Build optimization failed: found pages without a React Component as default export in `

- 해결 방안 : `pages` 아래 있는 페이지 컴포넌트 디렉터리 안에는 `style`을 정의하지 않고, 페이지 컴포넌트의 스타일은 `styles/pages/`에 정의합니다.

- 논의 내용: 단, 현재 `style`에 대한 경로 alias가 설정되어있지 않으므로, 이에 대한 설정이 필요합니다.
(관련 이슈) #93 

- 참고 자료:
https://velog.io/@yonghk423/Next.js-vercel-%EB%B0%B0%ED%8F%AC-%EC%97%90%EB%9F%AC

**Comment by Hyevvy** [comment]
멋져용
